larray/core.py

```diff
@@ -1335,7 +1337,7 @@ def __getitem__(self, key):
         """
         key is a label-based key (slice and fancy indexing are supported)
         """
-        return self.group(key)
+        return self._group(key)
```

Comment:
even if we do not deprecate all(), I would rather move the functionality here in getitem (and call getitem from _group) than the opposite. It's a good practice to make the deprecated function call the non-deprecated one.